astropy/nddata/mixins/ndslicing.py

```diff
@@ -11,29 +18,224 @@ class NDSlicingMixin(object):
     """
     Mixin to provide slicing on objects using the NDData interface.
 
-    When subclassing, be sure to list the superclasses in the correct order
-    so that the subclass sees NDData as the main superclass. See
-    `~astropy.nddata.NDDataArray` for an example.
+    The most common cases of the properties of `~astropy.nddata.NDData` are
+    covered, i.e. if uncertainty, mask and wcs are ``None`` or
+    `~numpy.ndarray` (see Notes for complete list of restrictions) but since
+    NDData deliberatly enforces nothing on most of it's properties there might
+    be a need to extend this. The `NDSlicingMixin` was designed to allow
+    subclasses to extend just the portion they need changing without having
+    to rewrite the whole Mixin (see Notes for detailed suggestions on how
+    to do that).
+
+    Notes
```

Comment:
I wonder if the bulk of the notes here should be added to the astropy documentation. The advice about subclassing is useful, but might be easier to find in the main documentation, with a reference here to those docs. I'm referring to this: https://github.com/astropy/astropy/blob/master/docs/nddata/subclassing.rst